src/app/portlets/dot-edit-page/layout/components/dot-theme-selector/dot-theme-selector.component.ts

```diff
@@ -0,0 +1,115 @@
+import { Observable } from 'rxjs/Observable';
+import { Site } from 'dotcms-js/dotcms-js';
+
+/**
+ * The DotThemeSelectorComponent is modal that
+ * show the themes of the available hosts
+ * @export
+ * @class DotThemeSelectorComponent
+ */
+@Component({
+    providers: [PaginatorService],
+    selector: 'dot-theme-selector',
+    templateUrl: './dot-theme-selector.component.html',
+    styleUrls: ['./dot-theme-selector.component.scss']
+})
+export class DotThemeSelectorComponent implements OnInit {
+    themes: DotTheme[];
+    @Input() value: DotTheme;
+    @Output() selected = new EventEmitter<DotTheme>();
+    @Output() close = new EventEmitter<boolean>();
+    @ViewChild('searchInput') searchInput: ElementRef;
+    @ViewChild('dataGrid') datagrid: DataGrid;
+
+    current: DotTheme;
+
+    constructor(public dotMessageService: DotMessageService, public paginatorService: PaginatorService) {}
+
+    ngOnInit() {
+        this.dotMessageService
+            .getMessages(['editpage.layout.theme.header', 'editpage.layout.theme.search', 'dot.common.apply', 'dot.common.cancel'])
+            .subscribe();
+        this.paginatorService.url = 'v1/themes';
+        this.paginatorService.paginationPerPage = 8;
+        this.current = this.value;
+        Observable.fromEvent(this.searchInput.nativeElement, 'keyup')
+            .debounceTime(500)
+            .subscribe((keyboardEvent: Event) => {
+                this.filterThemes(keyboardEvent.target['value']);
+            });
+    }
+
+    /**
+     * Load new page of themes.
+     * @param {LazyLoadEvent} event
+     *
+     * @memberof DotThemeSelectorComponent
+     */
+    paginate($event: LazyLoadEvent) {
+        this.paginatorService.getWithOffset($event.first).subscribe((items: DotTheme[]) => {
+            this.themes = items;
+            this.datagrid.first = $event.first;
+        });
+    }
+
+    /**
+     * Handle change in the host to load the corresponding themes.
+     * @param {Site} site
+     *
+     * @memberof DotThemeSelectorComponent
+     */
+    siteChange(site: Site) {
+        this.searchInput.nativeElement.value = null;
+        this.setPagination(site.identifier);
+        this.paginate({ first: 0 });
+    }
+
+    /**
+     * Set the selected Theme by the user while the modal is open.
+     * @param {DotTheme} theme
+     *
+     * @memberof DotThemeSelectorComponent
+     */
+    selectTheme(theme: DotTheme) {
+        this.current = theme;
+    }
+
+    /**
+     * Propagate the selected theme once the user apply the changes.
+     *
+     * @memberof DotThemeSelectorComponent
+     */
+    apply(): void {
+        this.selected.emit(this.current);
+    }
+
+    /**
+     * Propagate the close event wen the modal closes.
+     *
+     * @memberof DotThemeSelectorComponent
+     */
+    hideDialog(): void {
+        this.close.emit(false);
+    }
+
+    private filterThemes(searchCriteria?: string) {
+        this.setPagination(null, searchCriteria);
```

Comment:
I think is better call here directly

this.paginatorService.setExtraParams('searchParam', searchCriteria);

Reply:
you forgot this, I feel like we don't need the setPagination method

Reply:
@freddyucv, @alfredo-dotcms thoughts ?